client/app/tasks/components/TaskDetailsModal.tsx

```diff
@@ -0,0 +1,131 @@
+import { useState } from 'react'
+import { useRouter } from 'next/navigation'
+import { format } from 'date-fns'
+import { FiEdit, FiX, FiTrash } from 'react-icons/fi'
+import { MdAccountCircle } from 'react-icons/md'
+import {
+  Dialog,
+  DialogContent,
+  Drawer,
+  DrawerContent,
+  Button,
+} from '@/_components/ui'
+import { colorMap } from '@/_consts'
+import { useMediaQuery } from '@/_hooks'
+import { TaskTypeWithUser } from '@/_types'
+import { DeleteTaskModal } from './DeleteTaskModal'
+
+type TaskDetailsModalProps = {
+  selectedTask: TaskTypeWithUser | null
+  isModalOpen: boolean
+  onClose: () => void
+  onOpen: () => void
+}
+
+const TaskDetailsModal = ({
+  selectedTask,
+  isModalOpen,
+  onClose,
+  onOpen,
+}: TaskDetailsModalProps) => {
+  const router = useRouter()
+  const isDesktop = useMediaQuery('(min-width: 768px)')
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false)
+
+  if (!selectedTask) return null
+
+  const formattedDate = format(selectedTask.date, 'MMM/dd/yyyy')
+
+  const handleDeleteIcon = () => {
+    onClose()
+    setIsDeleteModalOpen(true)
+  }
+
+  const closeDeleteModal = (showTaskModal?: boolean) => {
+    setIsDeleteModalOpen(false)
+    if (showTaskModal) onOpen()
```

Comment:
Is there a case showTaskModal becomes true?

Reply:
Yes, when you click `Cancel` in delete modal, you want to show TaskDetailsModal